components/api/core/api_impl.bal

```diff
@@ -54,33 +53,53 @@ public function getTokens (http:Request getTokensReq) returns http:Response {
 
 public function createOrg(http:Request createOrgReq, gen:OrgCreateRequest createOrgsBody) returns http:Response {
     if (createOrgReq.hasHeader(constants:AUTHENTICATED_USER)) {
-        string userId = createOrgReq.getHeader(constants:AUTHENTICATED_USER);
-        var res = db:insertOrganization(userId, createOrgsBody);
-        if (res is error) {
-            log:printError("Unexpected error occured while inserting organization " + untaint createOrgsBody.orgName, err = res);
-            return buildUnknownErrorResponse();
+        boolean | error isMatch = createOrgsBody.orgName.matches("^[a-z0-9]+(-[a-z0-9]+)*$");
+        if (isMatch is boolean){
+            if (isMatch) {
+                log:printDebug(io:sprintf("\'%s\' is a valid organization name", createOrgsBody.orgName));
+                string userId = createOrgReq.getHeader(constants:AUTHENTICATED_USER);
+                var orgRes = db:insertOrganization(userId, createOrgsBody);
+                if (orgRes is error) {
+                    log:printError("Unexpected error occured while inserting organization " + untaint createOrgsBody.orgName, err = orgRes);
+                    return buildUnknownErrorResponse();
+                } else {
+                    log:printDebug(io:sprintf("New organization \'%s\' added to REGISTRY_ORGANIZATION. Author : %s", createOrgsBody.orgName, userId));
+                    return addOrgUserMapping(userId, createOrgsBody.orgName);
+                }
+            } else {
+                log:printError(io:sprintf("Insertion denied : \'%s\' is an invalid organization name", createOrgsBody.orgName));
+                return buildErrorResponse(http:METHOD_NOT_ALLOWED_405, constants:API_ERROR_CODE, "Unable to create organization",
+                                                "Organization name is not valid");
+            }
         } else {
-            http:Response createOrgRes = new;
-            createOrgRes.statusCode = http:OK_200;
-            log:printDebug(io:sprintf("Organization \'%s\' is created. Author : %s", createOrgsBody.orgName, userId));
-            return createOrgRes;
+            log:printError("Unable to create organization", err = isMatch);
+            return buildUnknownErrorResponse();
         }
     } else {
-        log:printError("Unauthenticated request. Username is not found");
-        return buildErrorResponse(http:UNAUTHORIZED_401, constants:API_ERROR_CODE, "Unable to create organization",
-                                  "Unauthenticated request. Auth token is not provided");
+        log:printError("Unauthenticated request : Username is not found");
+        return buildErrorResponse(http:UNAUTHORIZED_401, constants:API_ERROR_CODE, "Unable to create organization", 
+                                                            "Unauthenticated request. Auth token is not provided");
+    }
+}
+
+function addOrgUserMapping(string userId, string orgName) returns http:Response{
```

Comment:
Shall we pass the role here as well ? And set the role to "admin" by the caller ? This would be more generic then. (We can keep the set of roles in a constants)